docs/api/v2/node.md

```diff
@@ -53,11 +53,11 @@ method publish*(w: WakuNode, topic: Topic, message: WakuMessage)
   ##
   ## Status: Implemented.
 
-method query*(w: WakuNode, query: HistoryQuery): HistoryResponse
-  ## Queries for historical messages.
+method query*(w: WakuNode, query: HistoryQuery, handler: QueryHandlerFunc) {.async, gcsafe.} =
```

Comment:
I'm confused. Here you said you were happy with this signature https://github.com/status-im/nim-waku/issues/179#issuecomment-696672928 but this is different from the one in issue https://github.com/status-im/nim-waku/issues/179#issue-706075679

Reply:
@oskarth yeah, I realized we need to change it as the store protocol will receive multiple HistoryResponses due to it potentially sending the request to multiple nodes. Additionally it will need to be more async like the WakuFilter protocol as Kim had mentioned.

Reply:
Ok, it would be good if you updated in the issue then for clarity

Reply:
https://github.com/status-im/nim-waku/issues/179